craft_application/services/testing.py

```diff
@@ -83,3 +86,22 @@ def _get_backend(self) -> models.SpreadBackend:
             prepare_each=f'"$PROJECT_PATH"/spread/.extension backend-prepare-each {name}',
             restore_each=f'"$PROJECT_PATH"/spread/.extension backend-restore-each {name}',
         )
+
+    def _get_spread_executable(self) -> str:
+        """Get the executable to run for spread.
+
+        :returns: The spread command to use
+        :raises: CraftError if spread cannot be found.
+        """
+        # Spread included in the application.
+        if path := shutil.which(f"{self._app.name}.spread"):
+            return path
+        # Fall back to generic spread
```

Comment:
Maybe if craft.spread is missing we should just fail; there's a fair chance that whatever `spread` shows up in this `which()` call will be unsuitable for us (the versioning/release story there is not ideal)

Reply:
That's reasonable - if an app has a good reason to choose a different name it can override this method.